### ISSUE TITLE
kythe tool: display edge ordinals [phab:D1687]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2113]
_Created by **schroederc** at 2017-06-01 16:53:45_: 

Fixes #1679

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-06-01 17:01:16_:

Suggestion: `OrdinalKind` (since it's not an edge)

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-06-01 17:01:16_:

Can this check be done in the loop below?

```
if hasOrdinal || edge.Ordinal != 0 {
```
?

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2017-06-01 17:06:58_:

This is attempting to check if any edges in the group has an ordinal first.  This ensures that the 0th edge of a kind not caught by OrdinalEdge would be displayed properly if it has any further ordinals.

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-06-01 17:09:10_:

%s/OrdinalEdge/OrdinalKind/g

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-06-01 17:31:37_:

Hmm. There shouldn't be any ordinals on edges that don't satisfy that property. If the important property isn't the //expectation// of ordinals but their presence, might it be simpler to just check that for each?

```
_, ord, ok := ParseOrdinal(kind)
if ok {
  … do the ordinal thing…
} else {
  … don't …
}
```
?

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2017-06-01 17:35:15_:

`ParseOrdinal` won't work because the ordinals are not embedded as suffixes.

This really only affects `param` edges right now so technically the `OrdinalKind` check will be enough, but I'm guarding against custom edges (or new ordinal edge kinds that haven't made their way into the schema lib).

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-06-01 17:45:24_:

Bleh

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2017-06-01 17:47:59_

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2017-06-01 17:49:14_:

I've changed my mind.  Ensuring the 0th edge is displayed the same is not worth the effort.  It'll be obvious when it is displayed next to ".1", ".2", etc. edges.

**Comment by kythe** [comment]
_**schroederc** closed via commit 61f01fc486044dd22cd4f8305ce84550409895fd at 2017-06-01 17:49:57_
